### ISSUE TITLE
Support for scram-sha-256

### ISSUE THREAD

**Issue by joshuakb2** [dylex/postgresql-typed#28]
As far as I can tell, this library currently has no support for the newer SCRAM-SHA-256 authentication method, which is more secure than MD5. Are there any plans to support this?

**Comment by dylex** [comment]
That's true.  It looks like the protocol requires SASL (https://www.postgresql.org/docs/current/sasl-authentication.html). I don't see any up-to-date SASL implementations in haskell and am not sure what all it involves but I'm certainly open to contributions or pointers.
